include/alpaka/block/shared/st/BlockSharedMemStMember.hpp

```diff
@@ -0,0 +1,146 @@
+                        }
+
+                    public:
+                        //-----------------------------------------------------------------------------
+#ifndef NDEBUG
+                        BlockSharedMemStMemberImpl(uint8_t* mem, unsigned int capacity) : m_mem(mem), m_capacity(capacity) {}
+#else
+                        BlockSharedMemStMemberImpl(uint8_t* mem, unsigned int) : m_mem(mem) {}
+#endif
+                        //-----------------------------------------------------------------------------
+                        BlockSharedMemStMemberImpl(BlockSharedMemStMemberImpl const &) = delete;
+                        //-----------------------------------------------------------------------------
+                        BlockSharedMemStMemberImpl(BlockSharedMemStMemberImpl &&) = delete;
+                        //-----------------------------------------------------------------------------
+                        auto operator=(BlockSharedMemStMemberImpl const &) -> BlockSharedMemStMemberImpl & = delete;
+                        //-----------------------------------------------------------------------------
+                        auto operator=(BlockSharedMemStMemberImpl &&) -> BlockSharedMemStMemberImpl & = delete;
+                        //-----------------------------------------------------------------------------
+                        /*virtual*/ ~BlockSharedMemStMemberImpl() = default;
+
+                        template <typename T>
+                        void alloc() const
+                        {
+                            uint8_t* buf = &m_mem[m_allocdBytes];
+                            new (buf) T();
+                            m_allocdBytes += alignPitch<T>();
+#if (defined ALPAKA_DEBUG_OFFLOAD_ASSUME_HOST) && (! defined NDEBUG)
+                            ALPAKA_ASSERT(m_allocdBytes < m_capacity);
+#endif
+                        }
+
+#if BOOST_COMP_GNUC
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wcast-align" // "cast from 'unsigned char*' to 'unsigned int*' increases required alignment of target type"
+#endif
+                        template <typename T>
+                        T& getLatestVar() const
+                        {
+                           return *reinterpret_cast<T*>(&m_mem[m_allocdBytes-alignPitch<T>()]);
+                        }
+#if BOOST_COMP_GNUC
+    #pragma GCC diagnostic pop
+#endif
+
+                        void free() const
+                        {
+                            m_allocdBytes = 0u;
+                        }
+
+                    };
+                }
+                //#############################################################################
+                //! Static block shared memory provider using a pointer to
+                //! externally allocated fixed-size memory, likely provided by
+                //! BlockSharedMemDynMember.
+                template<unsigned int TDataAlignBytes = 4>
```

Comment:
Can we use `core::vectorization::defaultAlignment` instead of the hard coded 4 here? (note: `defaultAlignment` is in bits)

Reply:
`core::vectorization::defaultAlignment`, the smallest alignment defined https://github.com/alpaka-group/alpaka/blob/develop/include/alpaka/core/Vectorize.hpp#L51 is 16, which would not make much sense in bits.